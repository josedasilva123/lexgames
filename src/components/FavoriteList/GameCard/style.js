import styled from "styled-components";

//Estilos menos dinâmicos também podem ser comtemplados no 

export const StyledGameCard = styled.li`
    display: flex;
    align-items: center;
    justify-content: space-between;    
    gap: 2rem;
    padding: 16px;
    border: 1px solid var(--color-lightblue);
    border-radius: 4px;

    @media (max-width: 450px){
        align-items: flex-start;
        flex-direction: column;

        button{
            width: 100%;   
        }
    } 
`