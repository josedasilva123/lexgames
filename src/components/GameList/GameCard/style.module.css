.gameCard{
    display: flex;
    align-items: center;
    justify-content: space-between;
    gap: 2rem;
    padding: 16px;
    border: 1px solid var(--color-lightblue);
    border-radius: 4px;
}

@media (max-width: 450px){
    .gameCard{
        align-items: flex-start;
        flex-direction: column;
    }   
    .gameCard button{
        width: 100%;   
    }    
}
