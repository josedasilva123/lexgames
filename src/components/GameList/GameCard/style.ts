import styled from "styled-components";

export const StyledGameCard = styled.li`
  display: flex;
  flex-direction: column;
  gap: 1rem;
  justify-content: space-between;
  padding: 2rem;
  border: 1px solid var(--color-white);

  h3 {
    margin-top: 1rem;
    margin-bottom: 1rem;
    height: 46px;
  }
`;
