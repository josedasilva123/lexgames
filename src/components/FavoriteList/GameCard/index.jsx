import React from "react";
import { StyledButton } from "../../../styles/button";
import { StyledParagraph, StyledTitle } from "../../../styles/typography";
import { StyledGameCard } from "./style";

const GameCard = ({ game, removeGame }) => {
  return (
    <StyledGameCard>
      <div>
        <StyledParagraph>{game.genre}</StyledParagraph>
       
        <StyledTitle tag="h3" fontSize="two">
          {game.title}
        </StyledTitle>
      </div>
      <StyledButton buttonStyle="solid2" onClick={() => removeGame(game)}>Remover</StyledButton>
    </StyledGameCard>
  );
};

export default GameCard;
