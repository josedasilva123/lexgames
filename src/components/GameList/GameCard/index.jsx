import React from "react";
import { StyledTitle } from "../../../styles/typography";
import { StyledGameCard } from "./style";

const GameCard = ({ game, removeGame }) => {
  return (
    <StyledGameCard>
      <div>
        <span className="paragraph">{game.category}</span>
        <StyledTitle tag="h3" fontSize="two">
          {game.name}
        </StyledTitle>
      </div>
      <button className="btn solid2 default" onClick={() => removeGame(game)}>
        Remover
      </button>
    </StyledGameCard>
  );
};

export default GameCard;
