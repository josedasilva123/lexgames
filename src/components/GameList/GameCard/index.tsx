import React, { useContext } from "react";
import { GamesContext } from "../../../contexts/GamesContext/GamesContext";
import { iGame } from "../../../contexts/@types/types";
import { StyledButton } from "../../../styles/button";
import { StyledGameCard } from "./style";

interface iGameCardProps{
  game: iGame;
}

const GameCard = ({ game }: iGameCardProps) => {
  const { addGame } = useContext(GamesContext);
  return (
    <StyledGameCard>
      <div className="content">
        <img src={game.thumbnail} alt={game.title} />
        <h3>{game.title}</h3>
        <p>{game.short_description}</p>
      </div>
      <StyledButton buttonStyle="solid1" onClick={() => addGame(game)}>
        Adicionar
      </StyledButton>
    </StyledGameCard>
  );
};

export default GameCard;
