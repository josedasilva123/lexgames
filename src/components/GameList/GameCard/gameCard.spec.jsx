import { render, screen, waitFor } from "@testing-library/react"
import GameCard from "."

const mockGame = {
    thumbnail: "",
    title: 'League of Legends',
    short_description: 'Lorem ipsum'
}

const addGame = jest.fn();

describe("<GameCard />", () => {
    it("should render", () => {
        render(
            <GameCard game={mockGame} addGame={addGame} />            
        )

        const thumb = screen.getByRole("img");
        const heading = screen.getByText("League of Legends");
        const paragraph = screen.getByText("Lorem ipsum");
        const button = screen.getByRole("button");

        expect(thumb).toBeInTheDocument();
        expect(heading).toBeInTheDocument();
        expect(paragraph).toBeInTheDocument();
        expect(button).toBeInTheDocument();
    })

    it("should match with the snapshot", () => {
        const { container } = render(
            <GameCard game={mockGame} addGame={addGame} />    
        )
        expect(container).toMatchSnapshot();
    })
})