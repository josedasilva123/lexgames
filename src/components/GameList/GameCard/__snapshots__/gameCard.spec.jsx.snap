// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameCard /> should match with the snapshot 1`] = `
<div>
  <li
    class="sc-dkzDqf FDbzK"
  >
    <div
      class="content"
    >
      <img
        alt="League of Legends"
        src=""
      />
      <h3>
        League of Legends
      </h3>
      <p>
        Lorem ipsum
      </p>
    </div>
    <button
      class="sc-bczRLJ hoyupM"
    >
      Adicionar
    </button>
  </li>
</div>
`;
